tools/cli/NearObjectCliUwbSessionEventCallbacks.cxx

```diff
@@ -1,36 +1,54 @@
 
 #include <iostream>
-
+#include <iomanip>
 #include <nearobject/cli/NearObjectCliUwbSessionEventCallbacks.hxx>
+#include <notstd/tostring.hxx>
 
 using namespace nearobject::cli;
+using namespace strings::ostream_operators;
 
 void
-NearObjectCliUwbSessionEventCallbacks::OnSessionEnded(::uwb::UwbSession* /* session */, ::uwb::UwbSessionEndReason /* reason */)
+NearObjectCliUwbSessionEventCallbacks::OnSessionEnded(::uwb::UwbSession* session, ::uwb::UwbSessionEndReason reason)
 {
-    std::cout << "Session Ended" << std::endl;
+    std::cout << "Session with id="
+              << "0x" << std::hex << std::setw(8) << std::setfill('0') << session->GetId() << ": Session Ended" << std::endl;

```

Comment:
Instead of using `0x` directly, we should use the generic stream manipulator `showbase` instead. You'll also have to set the fill positioning with either `std::<left|right|internal>`. Here, `std::internal` makes sense since it will add the fill characters in between the base  (0x) and the value, eg. `0x******VV` where `*` is the fill char and `VV` is the value.

```suggestion
              << std::showbase << std::internal << std::hex << std::setw(8) << std::setfill('0') << session->GetId() << ": Session Ended" << std::endl;

```